packages/access-client/test/agent.test.js

```diff
@@ -23,31 +23,45 @@ describe('Agent', function () {
     const agent = await Agent.create()
     const space = await agent.createSpace('test-create')
 
-    assert(typeof space.did === 'string')
-    assert(space.proof)
+    assert(typeof space.did() === 'string')
   })
 
-  it('should add proof when creating acccount', async function () {
+  it('should add proof when creating account', async function () {
     const agent = await Agent.create()
     const space = await agent.createSpace('test-add')
+    const authorization = await space.createAuthorization(agent, {
+      access: { '*': {} },
+      expiration: Infinity,
+    })
+
+    await agent.importSpaceFromDelegation(authorization)
     const delegations = agent.proofs()
 
-    assert.equal(space.proof.cid, delegations[0].cid)
+    assert.equal(authorization.cid, delegations[0].cid)
   })
 
   it('should set current space', async function () {
     const agent = await Agent.create()
     const space = await agent.createSpace('test')
+    const authorization = await space.createAuthorization(agent, {
```

Comment:
I wouldn't mind if we left it so that `Agent#createSpace` automatically sets up authz, importing the space, and setting current, which would be a lot more back compat with what we've had. Not sure what's best here, but I'm worried that changing the behavior of `createSpace` too much might lead to headaches for end-users. I'd be inclined to try to keep all the current side effects for the same method (if easy), and only introduce new APIs/sideEffects in new methods that end-users can opt into over time.

Reply:
I think that was a really bad design choice because it effectively creates a space that is not usable and throws away the key so it never can be made usable. New API intentionally does not save anything to the agent so you have to arrange recovery beforehand.

I think doing breaking changes before we actually launch is probably better than shipping a footgun for the users, especially given that only users right now are us

<img width="709" alt="image" src="https://github.com/web3-storage/w3up/assets/21236/b3425596-a030-4e1e-8bfd-ee217e95226a">

And jchris who's been running into this problem and keeps asking us to fix it.

---

If you insist I will create another method and leave `createSpace` alone, but I think it would be a mistake given that our API is not yet stable, has issues and it's essentially use fixing our apps.